module-2/backend/week_four.md

```diff
@@ -10,14 +10,64 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 ### Questions
 
 * What is a cookie?
+
+  A cookie is a piece of data assigned to and saved in your browser when you visit a website
+
 * What’s the difference between a session and a cookie?
+
+  a session is encrypted, and can hold more data(?)
+
 * What’s a flash and when do you want to use flashes?
+
+  a flash is a hash that can be used to send temporary messages to a given view
+
 * Why do people say “HTTP is stateless”?
+
+  Because each response from a server doesn't carry 'state' from other responses
+
 * What’s authentication? Explain.
+
+   Authentication is what users to have unique experiences on a web app, e.g.
+   a logged-in has a more options than a non logged-in user.
```

Comment:
I would say the idea of `having more options` is probably more closely related to authorization. Authentication is primarily about knowing who a user is, not necessarily what they can do (though the two are tightly coupled).